OpenRA.Mods.Common/Traits/DockClientBase.cs

```diff
@@ -0,0 +1,54 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public abstract class DockClientBaseInfo : ConditionalTraitInfo, IDockClientInfo, Requires<DockClientManagerInfo> { }
+
+	public abstract class DockClientBase<InfoType> : ConditionalTrait<InfoType>, IDockClient, INotifyCreated where InfoType : DockClientBaseInfo
+	{
+		readonly Actor self;
+
+		public abstract BitSet<DockType> GetDockType { get; }
+		public DockClientManager DockClientManager { get; }
+
+		protected DockClientBase(Actor self, InfoType info)
+			: base(info)
+		{
+			this.self = self;
+			DockClientManager = self.Trait<DockClientManager>();
+		}
+
+		protected virtual bool CanDock()
+		{
+			return true;
+		}
+
+		public virtual bool DockingPossible(BitSet<DockType> type, bool forceEnter = false)
```

Comment:
`IsDockingPossible`. Boolean members should start with `is` or similar.
Also `CanDock` and `IsDockingPossible` seem redundant to each other.

Reply:
> IsDockingPossible. Boolean members should start with is or similar.

sure

> Also CanDock and IsDockingPossible seem redundant to each other.

They are completely different, you can read the documentation or just check out `DockClientBase`